energy-integration/common-modules/weekly-rewards-splitting/src/lib.rs

```diff
@@ -98,15 +100,15 @@ pub trait WeeklyRewardsSplittingModule:
         all_rewards
     }
 
-    fn claim_single<CollectRewardsFn: Fn(&Self, Week) -> TokenAmountPairsVec<Self::Api>>(
+    fn claim_single<WRSM: WeeklyRewardsSplittingTraitsModule<WeeklyRewardsSplittingMod = Self>>(
         &self,
+        wrapper: &WRSM,
         user: &ManagedAddress,
         current_week: Week,
-        collect_rewards_fn: CollectRewardsFn,
         claim_progress: &mut ClaimProgress<Self::Api>,
     ) -> PaymentsVec<Self::Api> {
         let total_rewards =
-            self.collect_and_get_rewards_for_week(claim_progress.week, collect_rewards_fn);
+            WRSM::collect_and_get_rewards_for_week_base(wrapper, self, claim_progress.week);
```

Comment:
Just call the method from wrapper instead of doing fully qualified call.
```suggestion
            wrapper.collect_and_get_rewards_for_week_base(self, claim_progress.week);
```

Reply:
Done.